src/Backend/Core/Installer/ModuleUninstaller.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Backend\Core\Installer;
+
+/*
+ * This file is part of Fork CMS.
+ *
+ * For the full copyright and license information, please view the license
+ * file that was distributed with this source code.
+ */
+
+/**
+ * The base-class for the uninstaller
+ */
+abstract class ModuleUninstaller extends ModuleInstaller
+{
+    use UninstallerTrait;
```

Comment:
First version of uninstaller suggested one class for installer and uninstaller (it is useful for micro modules). With a trait it can be implemented, for example:
```php
/**
 * Installer for the landing module
 */
class Installer extends ModuleInstaller implements InstallerInterface, UninstallerInterface
{

    use UninstallerTrait;

    public function install(): void
    {
        $this->addModule('Landing');

        $this->configureFrontendExtras();
    }

    public function uninstall(): void
    {
        $this->setModule('Landing');

        $this->dropModule();
    }

    private function configureFrontendExtras(): void
    {
        $this->insertExtra($this->getModule(), ModuleExtraType::widget(), 'Catalog', 'Catalog');
        $this->insertExtra($this->getModule(), ModuleExtraType::widget(), 'Text Columns', 'TextColumns');
        $this->insertExtra($this->getModule(), ModuleExtraType::widget(), 'Slider', 'Slider');
    }
}
```

Reply:
If you do it through inheritance, it turns out not so semantic:
```php
class Installer extends ModuleUninstaller implements InstallerInterface, UninstallerInterface
```

Reply:
A class should do one thing, so either install or uninstall, so I would just remove the trait and just add the code to the class

Reply:
Ok